cvmfs/catalog_mgr_impl.h

```diff
@@ -919,15 +956,20 @@ CatalogT *AbstractCatalogManager<CatalogT>::LoadFreeCatalog(
                                             const PathString     &mountpoint,
                                             const shash::Any     &hash)
 {
-  string new_path;
-  shash::Any check_hash;
-  const LoadError load_error = LoadCatalog(mountpoint, hash, &new_path,
-                                           &check_hash);
-  if (load_error != kLoadNew)
+  CatalogContext ctlg_context(hash, mountpoint, kCtlgLocationMounted);
+
+
+  const LoadReturn load_ret = LoadCatalogByHash(&ctlg_context);
+
+  // TODO(heretherebedragons) correct if statement?
+  if (load_ret != kLoadNew && load_ret != kLoadUp2Date) {
     return NULL;
-  assert(hash == check_hash);
+  }
+
+  assert(hash == ctlg_context.hash());  // TODO(heretherebedragons) why?
```

Comment:
Why not? If you think it's obvious / not needed anymore, please remove.
```suggestion
  assert(hash == ctlg_context.hash());  // TODO(heretherebedragons) why?
```

Reply:
not needed anymore. we do not change the hash. we just try to retrieve/download based on the hash given.